src/module.d.ts

```diff
@@ -0,0 +1,328 @@
+    TableHeadingCell: any;
+}
+
+class TableHeading extends React.Component<TableHeadingProps, any> {
+}
+
+export interface TableHeadingCellProps {
+    title?: string;
+    columnId?: number;
+    onClick?: Function;
+    onMouseEnter?: Function;
+    onMouseLeave?: Function;
+    icon?: any;
+    className?: string;
+    style?: React.CSSProperties;
+}
+
+class TableHeadingCell extends React.Component<TableHeadingCellProps, any> {
+}
+
+const TableContainer: (OriginalComponent: any) => any;
+
+} // namespace components
+
+export interface GriddleComponents {
+    Cell?: any;
+    CellContainer?: any;
+    ColumnDefinition?: any;
+    Row?: any;
+    RowContainer?: any;
+    RowDefinition?: any;
+    Table?: any;
+    TableBody?: any;
+    TableBodyContainer?: any;
+    TableHeading?: any;
+    TableHeadingContainer?: any;
+    TableHeadingCell?: any;
+    TableHeadingCellContainer?: any;
+    TableHeadingCellEnhancer?: any;
+    TableContainer?: any;
+    Layout?: any;
+    LayoutContainer?: any;
+    NextButton?: any;
+    NextButtonEnhancer?: any;
+    NextButtonContainer?: any;
+    NoResults?: any;
+    NoResultsContainer?: any;
+    PageDropdown?: any;
+    PageDropdownContainer?: any;
+    Pagination?: any;
+    PaginationContainer?: any;
+    PreviousButton?: any;
+    PreviousButtonEnhancer?: any;
+    PreviousButtonContainer?: any;
+    Filter?: any;
+    FilterEnhancer?: any;
+    FilterContainer?: any;
+    SettingsToggle?: any;
+    SettingsToggleContainer?: any;
+    SettingsWrapper?: any;
+    SettingsWrapperContainer?: any;
+    Settings?: any;
+    SettingsContainer?: any;
+}
+
+export interface GriddleEvents {
+    onFilter?: (filterText: string) => void;
+    onSort?: (sortProperties: any) => void;
+    onNext?: () => void;
+    onPrevious?: () => void;
+    onGetPage?: (pageNumber: number) => void;
+}
+
+export interface GriddleSortKey {
+    id: string;
+    sortAscending: boolean;
+}
+
+export interface GriddleStyleElements<T> {
+    Cell?: T;
+    Filter?: T;
+    Loading?: T;
+    NextButton?: T;
+    NoResults?: T;
+    PageDropdown?: T;
+    Pagination?: T;
+    PreviousButton?: T;
+    Row?: T;
+    RowDefinition?: T;
+    Settings?: T;
+    SettingsToggle?: T;
+    Table?: T;
+    TableBody?: T;
+    TableHeading?: T;
+    TableHeadingCell?: T;
+    TableHeadingCellAscending?: T;
+    TableHeadingCellDescending?: T;
+}
+
+export interface GriddleStyleIcons {
+    sortAscendingIcon?: any;
+    sortDescendingIcon?: any;
+}
+
+export interface GriddleStyleConfig {
+    classNames?: GriddleStyleElements<string>;
+    icons?: GriddleStyleElements<GriddleStyleIcons>;
+    styles?: GriddleStyleElements<React.CSSProperties>;
+}
+
+export interface GriddlePageProperties {
+    currentPage?: number;
+    pageSize?: number;
+    recordCount?: number;
+}
+
+interface RowRenderProperties {
+    rowKey?: string;
+    childColumnName?: string;
+    props?: {
+        children: components.ColumnDefinition[];
+    };
```

Comment:
I confess I can't quite figure out what the relationship should be between `renderProperties.rowProperties` and `RowDefinitionProps`, at least from a public API standpoint. Including `props` here, in particular, feels weird. I'm also not sure what to do about `childColumnName`, which has some logic implemented around it that ends up not changing anything.